### PR TITLE
feat: added YoloX types

### PR DESCRIPTION
**Pull request by benjats07**
This PR adds the capability to use output classes from YoloX.

In order to test you could run:
```
from unstructured_inference.inference.layout import DocumentLayout
from unstructured_inference.models.base import get_model
from unstructured.partition.pdf import partition_pdf, partition_pdf_or_image

f = "example-docs/reliance.pdf"

detectron_elements = partition_pdf(filename=f,strategy='hi_res',model_name='detectron2_onnx')
yolox_elements = partition_pdf(filename=f,strategy='hi_res',model_name='yolox')

types_detectron = set()
for element in detectron_elements:
        types_detectron.add(element.category)

types_yolox = set()
for element in yolox_elements:
        types_yolox.add(element.category)

types_yolox=sorted(list(types_yolox))
types_detectron=sorted(list(types_detectron))
print(f"Types detected by detectron: {types_detectron}")
print(f"Types detected by yolox: {types_yolox}")
```

The output will contain:
Types detected by detectron: ['FigureCaption', 'ListItem', 'NarrativeText', 'Table', 'Title', 'UncategorizedText']
Types detected by yolox: ['FigureCaption', 'Header', 'Image', 'ListItem', 'NarrativeText', 'UncategorizedText']

### CODE REVIEW

File: unstructured/documents/elements.py [hunk 1]
```diff
@@ -553,4 +561,12 @@ class Footer(Text):
     "Table": Table,
     "Header": Header,
     "Footer": Footer,
+    "Caption": Text,
```
Comment (cragwolfe) [inline unstructured/documents/elements.py:564]: should this not map to `FigureCaption` ?

File: unstructured/documents/elements.py [hunk 2]
```diff
@@ -553,4 +561,12 @@ class Footer(Text):
...
+    "Footnote": Footer,
+    "Formula": Formula,
+    "List-item": ListItem,
+    "Page-footer": Footer,
+    "Page-header": Header,  # Title?
```
Comment (cragwolfe) [inline unstructured/documents/elements.py:569]: good question in the comment. have any images available where yolox identified a `Page-header`?

File: unstructured/documents/elements.py (hunk 1 above)
Comment (benjats07) [reply unstructured/documents/elements.py:564]: You're right! Updating...

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,9 +1,9 @@
-## 0.10.12-dev2
+## 0.10.12-dev3
 
 ### Enhancements
 
 * Removed PIL pin as issue has been resolved upstream
-
+* YoloX types added
```
Comment (cragwolfe) [inline CHANGELOG.md:6]:
```suggestion
* YoloX types added

```

File: unstructured/documents/elements.py (hunk 2 above)
Comment (benjats07) [reply unstructured/documents/elements.py:569]: Actually no, I only found a lot of section-headers but any page-header 🤔

Comment (cragwolfe) [reply unstructured/documents/elements.py:569]: kk

**Comment by cragwolfe** [comment]
please provide test instructions in the description

**Comment by benjats07** [comment]
@cragwolfe Instructions added ✅

File: unstructured/documents/elements.py [hunk 4]
```diff
@@ -381,6 +381,14 @@ def to_dict(self) -> dict:
         return out
 
 
+class Formula(Element):
+    "An element containing formulas in a document"
+
+    category = "Formula"
```
Comment (badGarnet) [inline unstructured/documents/elements.py:387]: Not a blocking comment but just wonder what's an example of extracting formula? Like a math equation? Or something more general?

**Review by badGarnet — APPROVED** [review]

**Review by cragwolfe — APPROVED** [review]
Tested with: https://gist.github.com/cragwolfe/ed02fbfc7dbe2e0fc2feb92b1035bcaa#file-test-pr-element-labels-sh
after pip installing the latest `unstructured-inference`:

```
$ bash test-pr-element-labels.sh
unstructured_inference version is 0.5.21
Types detected by detectron: ['FigureCaption', 'NarrativeText', 'Title', 'UncategorizedText']
Types detected by yolox: ['NarrativeText', 'Table', 'UncategorizedText']
```

**Pull request merged**